plugins/graphiql/README.md

```diff
@@ -12,53 +12,64 @@ By exposing GraphiQL as a plugin instead of a standalone app, it's possible to p
 Start out by installing the plugin in your Backstage app:
 
 ```bash
+cd packages/app
 yarn add @backstage/plugin-graphiql
 ```
 
-Then add an entry to your App's `plugins.ts` to import the plugin.
-
-The plugin registers a `/graphiql` route, which you can link to from the Sidebar if desired.
+```diff
+# in packages/app/src/plugins.ts
++export { plugin as GraphiQL } from '@backstage/plugin-graphiql';
+```
 
-### Adding GraphQL endpoints
+```diff
+# in packages/app/src/App.tsx
++import { Router as GraphiQLRouter } from '@backstage/plugin-graphiql';
 
-For the plugin to function, you need to supply GraphQL endpoints through the GraphQLBrowse API, which is done by implementing the `GraphQLBrowseApi` exported by this plugin.
+ const AppRoutes = () => (
+   <Routes>
++    <Route path="/graphiql" element={<GraphiQLRouter />} />
+```
 
-If all you need is a static list of endpoints, the plugin exports a `GraphQLEndpoints` class that implements the `GraphQLBrowseApi` for you. Here's and example of how you could expose two GraphQL endpoints in your App:
+### Adding GraphQL endpoints
 
-```tsx
-import {
-  graphQlBrowseApiRef,
-  GraphQLEndpoints,
-} from '@backstage/plugin-graphiql';
+For the plugin to function, you need to supply GraphQL endpoints through the GraphQLBrowse API. This is done by implementing the `graphQlBrowseApiRef` exported by this plugin.
 
-// Implement the Graph QL browse API using a static list of endpoints
-const graphQlBrowseApi = GraphQLEndpoints.from([
-  // Use the .create function if all you need is a static URL and headers.
-  GraphQLEndpoints.create({
-    id: 'gitlab',
-    title: 'GitLab',
-    url: 'https://gitlab.com/api/graphql',
-    // Optional extra headers
-    headers: { Extra: 'Header' },
-  }),
-  {
-    id: 'hooli-search',
-    title: 'Hooli Search',
-    // Custom fetch function, this one is equivalent to using GraphQLEndpoints.create()
-    // with url set to https://internal.hooli.com/search
-    fetcher: async (params: any) => {
-      return fetch('https://internal.hooli.com/search', {
-        method: 'POST',
-        headers: { 'Content-Type': 'application/json' },
-        body: JSON.stringify(params),
-      }).then(res => res.json());
-    },
-  },
-]);
+If all you need is a static list of endpoints, the plugin exports a `GraphQLEndpoints` class that implements the `GraphQLBrowseApi` for you. Here's an example of how you could expose two GraphQL endpoints in your App:
 
-// ApiRegistry builder created somewhere in your App
-const builder = ApiRegistry.builder();
+```diff
```

Comment:
Isn't it a bit nicer to keep this as `ts`, so we get syntax highlighting?

Reply:
Specifically in this file, I felt that any drawback of missing highlighting was outweighed by the benefits of having it on diff form - to be able to see surrounding code and the precise lines to insert, where - without having to write a lot of free form text "in the x method of the y class under the z" etc :)